### PR TITLE
Send notifications to users who need to restage apps

### PR DESCRIPTION
**Pull request by jcscottiii**
## Goal
Addresses this card: https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6450
This PR will look for the apps that need to be restaged, then group the apps to individual owners then send a notification to each owner with their list of the apps.

### How does this meet the ACs
- AC: `There exists a way to generate a list of apps that need to be updated given that the app is using an older version of a given buildpack.`
  - Look at the `findOutdatedApps` function

- AC: `There exists a way to generate a list of Space Developers and Space Managers e-mails that are responsible for the app that needs to be updated.`
  - Look at the `findOwnersOfApps` function

### Testing the result of the ACs

- Run the integration test, which is documented in the README. 

#### Add getters for specific space roles instead of using space.Roles() in the go library.
We had to move from using `space.Roles()` to calling `/v2/spaces/guid/managers` and `/v2/spaces/guid/developers` manually because of a boolean issue when running this locally with PCFDev. More about it here: https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6455
You can see the work we had to do in the cf.go file (which should be ideally moved to the cf go library)
Issue for [Go library](https://github.com/cloudfoundry-community/go-cfclient): TBD

#### Add metadata for entities in the returned objects..
We need the metadata for the time stamp info. But the meta data doesn't get returned.
You can see the work we had to do in the cf.go file (which should be ideally moved to the cf go library)
Issue for [Go library](https://github.com/cloudfoundry-community/go-cfclient): TBD

## What this PR isn't / what it needs but will not be addressed in this PR.
More for the next PR that addresses this card https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-6455
Add tests for e-mail sending functionality
- Similar to https://github.com/18F/cg-dashboard/blob/master/mailer/mailer_test.go
- And https://github.com/18F/cg-dashboard/blob/master/helpers/testhelpers/docker/docker.go#L92
- This one requires pulling in dockertest https://github.com/ory/dockertest

### CODE REVIEW

**Comment by fureigh** [comment]
@jcscottiii, I'm taking a look now. Does it make sense to split any of this into a follow-up ticket or nah?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,6 +8,21 @@ We encourage you to read this project's CONTRIBUTING policy (you are here), its
 
 If you have any questions or want to read more, check out the [18F Open Source Policy GitHub repository](https://github.com/18f/open-source-policy), or just [shoot us an email](mailto:18f@gsa.gov).
 
+## Testing
+
+### Template tests
+
+The template tests are useful to see what a completed template looks like. When developing, you
+should add your expected result file in the `testdata` folder. The file should be a `.html` file
```
Comment (fureigh) [inline CONTRIBUTING.md]: @jcscottiii Directly in the `testdata` folder, or is there a recommended organization scheme? I see that the current example is `testdata/mail/notify/single_app.html`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,6 +8,21 @@ We encourage you to read this project's CONTRIBUTING policy (you are here), its
...
+so that when a user opens it, it will open in their browser.
```
Comment (fureigh) [inline CONTRIBUTING.md]: This is a nitpick, but: `browser` --> `web browser` (as opposed to their file browser).

File: main.go [hunk 3]
```diff
@@ -115,16 +137,63 @@ func isAppUsingOutdatedBuildpack(app cfclient.App, buildpack *cfclient.Buildpack
 	return timeOfLastBuildpackUpdate.After(timeOfLastAppRestage)
 }
 
-func findOutdatedApps(apps []cfclient.App, buildpacks map[string]cfclient.BuildpackResource) {
+func findOwnersOfApps(apps []cfclient.App) map[string][]cfclient.App {
+	// Mapping of users to the apps.
+	owners := make(map[string][]cfclient.App)
+	for _, app := range apps {
+		// Get the space
+		space, err := app.Space()
```
Comment (jmcarp) [inline main.go]: If multiple outdated apps live in the same space, we'll look up the same space again for each app. Same goes for space roles. WDYT about caching both to cut down on api calls?

Comment (jcscottiii) [reply main.go]: @jmcarp great idea! easy fix :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (jcscottiii) [reply CONTRIBUTING.md]: I guess it would be more like `testdata/{medium}/{feature}/{test_name}.html

Comment (jcscottiii) [reply CONTRIBUTING.md]: done below

**Pull request merged**